src/coreclr/vm/gcinfodecoder.cpp

```diff
@@ -2176,6 +2176,64 @@ template <typename GcInfoEncoding> void TGcInfoDecoder<GcInfoEncoding>::ReportRe
 
 #endif // Unknown platform
 
+#ifdef FEATURE_INTERPRETER
+template <> OBJECTREF* TGcInfoDecoder<InterpreterGcInfoEncoding>::GetStackSlot(
+                        INT32           spOffset,
+                        GcStackSlotBase spBase,
+                        PREGDISPLAY     pRD
+                        )
+{
+    OBJECTREF* pObjRef = NULL;
+
+    if( GC_SP_REL == spBase )
+    {
+        _ASSERTE(!"GC_SP_REL is invalid for interpreter frames");
+    }
+    else if( GC_CALLER_SP_REL == spBase )
+    {
+        _ASSERTE(!"GC_CALLER_SP_REL is invalid for interpreter frames");
+    }
+    else
+    {
+        _ASSERTE( GC_FRAMEREG_REL == spBase );
+
+        // HACK: The register slot we stash the frame pointer into varies per-architecture.
+        // CONTEXTGetFp is unavailable here (including its header doesn't work either).
+        uint8_t* fp = (uint8_t *)
+#if defined(TARGET_AMD64)
+        pRD->pCurrentContext->Rbp;
+#elif defined(TARGET_ARM)
+        // FIXME: CONTEXTGetFp says R7 but cgencpu.h says R11
```

Comment:
R11 is correct. I'll look into the `CONTEXTGetFp` implementation, it dates back 9 years and it always had the wrong register.

Reply:
The distinction boils down to the Thumb instruction set where only limited number of registers was accessible and R7 was indeed used as the frame pointer since PUSH instruction could only access R0-R7. I'll need to go recheck my assumptions about what is actually used. We do use Thumb code for both JIT and the native runtime. If clang/gcc use R7 for frame pointer, then `CONTEXTGetFp` would be correct for native code. JIT does use R11, as described both in clr-abi.md and `REG_FPBASE` in targetarm.h.